### PR TITLE
Add: Detention Camp

### PR DESCRIPTION
**Pull request by commy2**
For worldbuilder.

### CODE REVIEW

**Review by xezon — COMMENTED** [review]
## Door behaviour

Door moves down, but there is another door behind it.

shot_20230809_172156_1

## Spawn behaviour

Dozer just pops into existence on the pad. Does not drive out.

shot_20230809_172158_2

## Ranger spawn

Prison spawns 8 Rangers on destruction. Is a lot. Barracks spawns 2, Strategy Center spawns 4, War Factory spawns 5.

shot_20230809_172639_4

## Price

Detention Camp costs 1000. GLA Prison costs 1500. As designed?

**Comment by commy2** [comment]
Same prices would be boring. (AD $2500 vs WF $2000, Supply Stash $1500, Supply Center $2000 etc.)

$1000 for Det Camp is from CCG.

**Comment by commy2** [comment]
> Door moves down, but there is another door behind it.

Would require model edits

**Review by xezon — APPROVED** [review]
Debug needs to be removed before merge.

I believe 4 Rangers spawned on death are still too much.

Because 4 Rangers are worth 900. Structure costs 1000. This is 90% value. Other USA structures have less.

Barracks 75%
War Factory 56%
Strategy Center 36%

**Comment by commy2** [comment]
> Other USA structures have less.

Command Center: 112.5%
xD

File: Patch104pZH/GameFilesEdited/Data/INI/Object/FactionBuilding.ini [hunk 1]
```diff
@@ -7402,6 +7402,786 @@ Object AmericaStrategyCenter
...
+  End
+  Behavior            = DestroyDie ModuleTag_10
+    ;<NO DATA>
+  End
+  Behavior            = CreateObjectDie ModuleTag_12
+    CreationList      = OCL_ABPowerPlantExplode
+  End
+
+  Behavior            = CreateObjectDie ModuleTag_13
+    CreationList      = OCL_AmericanRangerDebris04
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/FactionBuilding.ini:8153]:
```suggestion
    CreationList      = OCL_AmericanRangerDebris03
```
Interestingly, it does spawn 8 in CCG while costing $1000: https://github.com/TheSuperHackers/GeneralsGamePatch/blob/main/Patch104pZH/GameFilesOriginalCCG/Data/INI/Object/FactionBuilding.ini#L5657-L5660

File: Patch104pZH/Design/Changes/v1.0/2183_detention_camp.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+date: 2022-07-31
+
+title: Adds unused Detention Camp
```
Comment (commy2) [inline Patch104pZH/Design/Changes/v1.0/2183_detention_camp.yaml]:
```suggestion
title: Adds unused USA Detention Camp to World Builder
```

File: Patch104pZH/Design/Changes/v1.0/2183_detention_camp.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+changes:
+  - fix: Add Detention Camp to world builder.
```
Comment (commy2) [inline Patch104pZH/Design/Changes/v1.0/2183_detention_camp.yaml]:
```suggestion
  - feature: The USA Detention Camp can now be placed in the world builder.
```

File: Patch104pZH/Design/Changes/v1.0/2183_detention_camp.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+date: 2022-07-31
```
Comment (commy2) [inline Patch104pZH/Design/Changes/v1.0/2183_detention_camp.yaml]:
```suggestion
date: 2023-07-31
```

**Pull request merged**